pmlc/dialect/pxa/analysis/strides.cc

```diff
@@ -183,4 +221,46 @@ Optional<StrideInfo> computeStrideInfo(pmlc::dialect::pxa::AffineReduceOp op) {
                            op.idxs());
 }
 
+StrideArray::StrideArray(unsigned numDims, int64_t offset)
+    : offset(offset), strides(numDims) {}
+
+StrideArray &StrideArray::operator*=(int64_t factor) {
+  offset *= factor;
+  for (auto &dim : strides)
+    dim *= factor;
+  return *this;
+}
+
+StrideArray &StrideArray::operator+=(const StrideArray &rhs) {
+  assert(strides.size() == rhs.strides.size() && "strides sizes much match");
+  offset += rhs.offset;
+  for (unsigned i = 0, e = strides.size(); i < e; ++i) {
+    strides[i] += rhs.strides[i];
+  }
+  return *this;
+}
+
+void StrideArray::print(raw_ostream &os) {
+  os << offset << ":[";
+  for (auto item : llvm::enumerate(strides)) {
+    if (item.index())
+      os << ", ";
+    os << item.value();
+  }
+  os << ']';
+}
+
+Optional<StrideArray> computeStrideArray(AffineMap map) {
+  std::vector<SmallVector<int64_t, 8>> flat;
+  if (failed(getFlattenedAffineExprs(map, &flat, nullptr)))
```

Comment:
I think this will do something incorrect if there are any ceildivs, etc.  I read the related code, and it looks like the class is uses (SimpleAffineExprFlattener) adds 'locals' as needed to handle modulus, etc, but then the function call that constructs the class ignores the induced locals.  So I think it will silently produce garbage data (probably an oversight?). Might be worth a test.  The correct behavior for use is to fail on ceildiv, etc.